src/main/java/model/Order.java

```diff
@@ -74,21 +83,96 @@ public double getTotalPrice() {
+     *
+     * @param headerBuilder the StringBuilder to append the header to
+     * @return the formatted receipt header
+     */
+    private StringBuilder getReceiptHeader(StringBuilder headerBuilder) {
+        return headerBuilder.append(ROW_DELIMITER)
+                .append(TITLE)
+                .append(ROW_DELIMITER)
 
-        receiptBuilder.append(ROW_DELIMITER)
+                .append("| ")
                 .append(restaurantName)
-                .append("\n")
+                .append(" ".repeat(ROW_DELIMITER.length() - restaurantName.length() - SUBTITLE_OFFSET))
+                .append("|\n")
+
+                .append("| ")
                 .append(restaurantAddress)
-                .append("\n")
-                .append(ROW_DELIMITER)
-                .append(TITLE)
-                .append(ROW_DELIMITER)
+                .append(" ".repeat(ROW_DELIMITER.length() - restaurantAddress.length() - SUBTITLE_OFFSET))
+                .append("|\n")
+
+                .append("| Order Type: ")
                 .append(orderType)
+                .append(" ".repeat(ROW_DELIMITER.length() - orderType.length() - ORDER_TYPE_OFFSET))
+                .append("|\n")
+
+                .append("| Order ID: ")
+                .append(orderID)
+                .append(" ".repeat(ROW_DELIMITER.length() - orderID.length() - ORDER_ID_OFFSET))
+                .append("|\n")
+
+                .append(ROW_DELIMITER)
                 .append(HEADERS)
                 .append(ROW_DELIMITER);
+    }
+
+    /**
+     * Returns a formatted receipt with summary (bottom part of the receipt) appended
+     *
+     * @param receiptBuilder the StringBuilder to append the summary to
+     * @param discount the discount to be applied to the order
+     * @return the formatted receipt with summary
+     */
+    private StringBuilder getReceiptSummary(StringBuilder receiptBuilder, double discount) {
+        double netTotal = getTotalPrice() / (1 + GST) / (1 + SERVICE_CHARGE) * (1 - discount);
+        double discountAmount = (discount) * netTotal / (1 - discount);
+        double serviceCharge = netTotal * SERVICE_CHARGE;
+        double gst = netTotal * (1 + SERVICE_CHARGE) * GST;
+        double grandTotal = netTotal + serviceCharge + gst;
+
+        receiptBuilder.append(ROW_DELIMITER)
+                .append(String.format((discount < SINGLE_DIGIT_THRESHOLD ?
+                        SINGLE_DIGIT_DISCOUNT_FORMAT : DISCOUNT_AMOUNT_FORMAT)
+                            + chooseFormat(discount) + " |\n", discount * 100,
+                            "Off Discount:", discountAmount))
+                .append(String.format(CHARGE_FORMAT + chooseFormat(netTotal) + " |\n",
+                        "Subtotal:", netTotal))
+                .append(ROW_DELIMITER)
+                .append(String.format(CHARGE_FORMAT + chooseFormat(serviceCharge) + " |\n",
+                        "Service Charge (" + (SERVICE_CHARGE * 100) + "%):", serviceCharge))
```

Comment:
Is the formatting for the discount and service charge intentionally not pushed to the far right? Also, does it handle 0% discount correctly?